### PR TITLE
Rewrite rrdtool_graph

### PR DESCRIPTION
**Pull request by murrant**
#### Please note

> Please read this information carefully.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

Just use a plain exec call as we are only making one rrdtool command and then exiting.
A long running rrdtool command causes more issues then not.
Attempt to gracefully close rrdtool -
Possible better fix to #4170

### CODE REVIEW

File: includes/rrdtool.inc.php [hunk 1]
```diff
@@ -88,9 +88,13 @@ function rrdtool_terminate() {
  * @param  array $rrd_pipes
  * @return integer
  */
-function rrdtool_pipe_close($rrd_process, &$rrd_pipes)
+function rrdtool_pipe_close(&$rrd_process, &$rrd_pipes)
```
Comment (laf) [inline includes/rrdtool.inc.php]: The & was what was removed last year to resolve some zombie processes - any reason to re-introduce it?

**Comment by allano337** [comment]
once patch was installed I had no issues with time outs adding & editing 8 new devices. Uninstalled the patch, was able to add & edit 2 new devices with no timeouts? hmm

File: includes/rrdtool.inc.php (hunk 1 above)
Comment (murrant) [inline includes/rrdtool.inc.php]: I misread that change as reverse, I will remove it.

**Comment by mmarchand** [comment]
it seems to fix the zombie processes but it also breaks some device bits graphs (the per device global traffic graph) when there are too many ports it seems (might be a limitation of shell_exec and command length maybe ...?)

**Comment by laf** [comment]
@mmarchand mind testing this add on patch to this PR as well please:

https://www.irccloud.com/pastebin/FJQoCeZz/

**Pull request closed without merging**